### ISSUE TITLE
[Bug] Collection fields validation

### ISSUE THREAD

**Issue by andreasvirkus** [pruvious/pruvious#58]
I'm getting a `422 - Unprocessable entity` when trying to update a custom collection item, but it seems to grab fields from other blocks.

For example, the props for the Testimonials block are the following, but the error claims to check for `alt` and `image` fields as well 🤔 

```js
const props = defineProps({
  title: textField({}),
  testimonials: repeaterField({
    required: true,
    subfields: {
      from: textSubfield(),
      company: textSubfield(),
      text: editorSubfield({
        required: true,
        toolbar: ['bold', 'italic', 'paragraph', 'link', 'bulletList'],
      }),
    },
  }),
})
```

![Screenshot 2024-09-19 at 09 40 31](https://github.com/user-attachments/assets/b1f9ed11-5105-4d9d-abf6-dbb71e60e533)

**Comment by murisceman** [comment]
Hey! Could you also dump the payload that is sent to the server?

It seems that the first item in the `testimonials` array contains excess data. Does recreating the repeater item resolve the issue?

**Comment by andreasvirkus** [comment]
It sometimes/sorta does, but it's flaky. For instance, when I recreated the repeater field for `testimonials`, the `ProjectsSlider` now claims to have testimonial block's fields:

![Screenshot 2024-09-21 at 07 52 12](https://github.com/user-attachments/assets/05326b1f-1bda-4b95-ae34-915b4e832ca0)

Whereas its definition is actually:

```js
defineProps({
  title: textField({
    description: 'The title of the portfolio block',
  }),
  list: repeaterField({
    description: 'List of projects to highlight',
    subfields: {
      project: recordSubfield({
        collection: 'projects',
        fields: {
          headline: true,
          shortDescription: true,
          tags: true,
          image: true,
          path: true,
        },
        populate: true,
        required: true,
        placeholder: 'Select a project',
      }),
    },
  }),
})
```

**Comment by murisceman** [comment]
Thank you for the additional info! I think I have identified the issue. It should now be fixed in `v3.12.9`.

**Comment by andreasvirkus** [comment]
Thanks for the quick fix! ✅
